booster/library/Booster/JsonRpc.hs

```diff
@@ -582,19 +587,31 @@ execResponse req (d, traces, rr) unsupported = case rr of
                 xs@(_ : _) -> Just xs
 
 toExecState ::
-    Pattern -> [Syntax.KorePattern] -> Maybe UniqueId -> RpcTypes.ExecuteState
-toExecState pat unsupported muid =
+    Pattern ->
+    [Syntax.KorePattern] ->
+    Maybe UniqueId ->
+    Maybe Predicate ->
+    Maybe (Map Variable Term) ->
+    RpcTypes.ExecuteState
+toExecState pat unsupported muid mrulePredicate mruleSubst =
```

Comment:
Are there any cases where only one or two `Maybe` are filled?
Probably we could add a record type to structure this better... 
```suggestion
    Pattern ->
    [Syntax.KorePattern] ->
    Maybe RuleApplication ->
    RpcTypes.ExecuteState
toExecState pat unsupported mRuleApp =...

data RuleApplication = 
    RuleApp
        { uid :: UniqueId
        , rulePred :: Predicate
        , ruleSubst :: Map Variable Term
        }
```